gov/politeia/proposals.go

```diff
@@ -90,371 +87,509 @@ func NewProposalsDB(politeiaURL, dbPath string) (*ProposalDB, error) {
-// returns the result.
-func (db *ProposalDB) proposal(searchBy, searchTerm string) (*pitypes.ProposalInfo, error) {
-	var pInfo pitypes.ProposalInfo
-	err := db.dbP.Select(q.Eq(searchBy, searchTerm)).Limit(1).First(&pInfo)
+// fetchProposalsData returns the parsed vetted proposals from politeia
+// API's. It cooks up the data needed to save the proposals in stormdb. It
+// first fetches the proposal details, then comments and then vote summary.
+// This data is needed for the information provided in the dcrdata UI. The
+// data returned does not include ticket vote data.
+func (db *ProposalsDB) fetchProposalsData(tokens []string) ([]*pitypes.ProposalRecord, error) {
+	// Fetch record details for each token from the inventory
+	recordDetails, err := db.fetchRecordDetails(tokens)
 	if err != nil {
-		log.Errorf("Failed to fetch data from Proposals DB: %v", err)
 		return nil, err
 	}
 
-	return &pInfo, nil
-}
+	// Fetch comments count for each token from the inventory
+	cr, err := db.client.CommentCount(commentsv1.Count{
+		Tokens: tokens,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Pi client CommentCount err: %v", err)
+	}
+	commentsCounts := cr.Counts
 
-// LastProposalsSync returns the last time a sync to update the proposals was run
-// but not necessarily the last time updates were synced in proposals.db.
-func (db *ProposalDB) LastProposalsSync() int64 {
-	return atomic.LoadInt64(&db.lastSync)
-}
+	// Fetch vote summary for each token from the inventory
+	sr, err := db.client.TicketVoteSummaries(ticketvotev1.Summaries{
+		Tokens: tokens,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Pi client TicketVoteSummaries err: %v", err)
+	}
+	voteSummaries := sr.Summaries
 
-// CheckProposalsUpdates updates the proposal changes if they exist and updates
-// them to the proposal db.
-func (db *ProposalDB) CheckProposalsUpdates() error {
-	if db == nil || db.dbP == nil {
-		return errDef
+	var proposals []*pitypes.ProposalRecord
+
+	// Iterate through every record and feed data used by dcrdata
+	for _, record := range recordDetails {
+		proposal := pitypes.ProposalRecord{}
+
+		// Record data
+		proposal.State = record.State
+		proposal.Status = record.Status
+		proposal.Version = record.Version
+		proposal.Timestamp = uint64(record.Timestamp)
+		proposal.Username = record.Username
+		proposal.Token = record.CensorshipRecord.Token
+
+		// Proposal metadata
+		pm, err := proposalMetadataDecode(record.Files)
+		if err != nil {
+			return nil, fmt.Errorf("proposalMetadataDecode err: %v", err)
+		}
+		proposal.Name = pm.Name
+
+		// User metadata
+		um, err := userMetadataDecode(record.Metadata)
+		if err != nil {
+			return nil, fmt.Errorf("userMetadataDecode err: %v", err)
+		}
+		proposal.UserID = um.UserID
+
+		// Comments count
+		proposal.CommentsCount =
+			int32(commentsCounts[record.CensorshipRecord.Token])
+
+		// Vote summary data
+		summary := voteSummaries[proposal.Token]
+		proposal.VoteStatus = summary.Status
```

Comment:
Let's assume the API can serve bad data.  Check like

```go
summary, found := voteSummaries[proposal.Token]
if !found {
    <log and continue>
```

I suspect zero values for all the fields would be problematic.